src/crawlee/storages/_key_value_store.py

```diff
@@ -52,6 +54,10 @@ class KeyValueStore(BaseStorage):
     ```
     """
 
+    # Cache for persistent (auto-saved) values
+    _cache: ClassVar[dict[str, dict[str, Any]]] = {}
```

Comment:
This way the cache will be shared between the default key-value store and its potential named counterparts, which means it will behave quite unpredictably if you use named key-value stores in your crawler.

I know you did this so that the cache would be shared between separate crawler instances, but that should work nonetheless - `KeyValueStore.open()` should always return the same instance.

Reply:
I was moving this cache to the class level. Because when using `self.get_key_value_value_value_store()` I was getting different instances of `KeyValueStore`. So different handlers had different cache state, this became noticeable after adding the `test_context_handlers_use_state` test.

But I don't see this behavior anymore and the test passes correctly, this is interesting.....